### PR TITLE
refactor: use create-jest-runner

### LINKED ISSUE #20: Support Jest's `runInBand` option
Was looking at trying to debug #19, but realised that the runner always uses a worker-farm and ignores Jest's `runInBand` option - this makes it hard to debug the runner.

Might look at adding this option first :)

### LINKED ISSUE #23: Switch from `worker-farm` to `jest-worker`?
In Jest 22 the team exposed the internal `worker-farm` equivalent, [`jest-worker`](https://github.com/facebook/jest/tree/master/packages/jest-worker). Would it be worth switching from `worker-farm` to `jest-worker` for consistency with Jest's default runner?

### PR DESCRIPTION
**Pull request by SimenB**
Closes #32
Closes #36
Fixes #23
Fixes #20

### CODE REVIEW

File: src/runESLint.js [hunk 1]
```diff
@@ -1,100 +1,42 @@
...
-    tests: [
-      {
-        duration: end - start,
-        testPath,
-        errorMessage,
-      },
-    ],
-    jestTestPath: testPath,
-  });
+const runESLint = ({ testPath, config }) => {
```
Comment (SimenB) [inline src/runESLint.js]: I recommend using https://github.com/jest-community/jest-runner-eslint/pull/37/files?w=1 here

File: integrationTests/__snapshots__/failing.test.js.snap [hunk 2]
```diff
@@ -6,7 +6,7 @@ exports[`Works when it has failing tests 1`] = `
   4:3  error  'hello' is not defined  no-undef
   5:3  error  'bye' is not defined    no-undef
 ✖ 2 problems (2 errors, 0 warnings)
-  ✕ ESLint
+  ✓ ESLint
```
Comment (SimenB) [inline integrationTests/__snapshots__/failing.test.js.snap]: not sure about this one... @rogeliog thoughts?

Comment (ljharb) [reply integrationTests/__snapshots__/failing.test.js.snap]: This should definitely be a blocker; the check shouldn’t appear if there are any failures.

File: integrationTests/__snapshots__/format.test.js.snap [hunk 3]
```diff
@@ -16,7 +16,7 @@ mocked-stack-trace  3 | if (a === 2) {
   6 | }
   7 |
 2 errors found.
-  ✕ ESLint
+  ✓ ESLint
```
Comment (ljharb) [inline integrationTests/__snapshots__/format.test.js.snap]: Also here

File: integrationTests/__snapshots__/failing.test.js.snap (hunk 2 above)
Comment (SimenB) [reply integrationTests/__snapshots__/failing.test.js.snap]: yup. it was a bug in my port, fixed now 🙂

**Comment by ljharb** [comment]
This should be a patch, but it’s a risky one, so whoever merges and releases it, please keep an eye out for failed greenkeeper issues popping up.

**Comment by SimenB** [comment]
I'd say it's a minor (because of added support for run in band), but I agree it comes with some risk.

I don't use this runner, could someone who do try to run it on their code base as a sanity check?

**Comment by ljharb** [comment]
Ah true, minor is good.

I’m traveling this week but will try it on Airbnb’s codebase on Monday if it’s not merged by then.

**Comment by SimenB** [comment]
@ljharb did you get a chance to test it out?

**Comment by ljharb** [comment]
aah, sorry, no i didn’t yet. I’ll try to make some time to do it next week!

**Comment by ljharb** [comment]
My work laptop's died on me, so I'm not sure when I'll be able to test this PR :-/

I don't think that needs to block it, though.

**Comment by rogeliog** [comment]
I just published it under `0.5.0`

**Pull request merged**